helix-term/src/ui/editor.rs

```diff
@@ -809,11 +810,18 @@ impl EditorView {
 
         let mut execute_command = |command: &commands::MappableCommand| {
             command.execute(cxt);
+            helix_event::dispatch(PostCommand { command, cx: cxt });
```

Comment:
I agree it's better to separate these two cases and that was also my intention.


insert_char is special cased [here](https://github.com/helix-editor/helix/blob/40959bb449ec9dfbff9249332853a02b50f483fe/helix-term/src/ui/editor.rs#L864) and can not trigger this codepath. `insert_char` takes an argument (a character) so it doesn't fit the signature of `MappableCommand` so its not a normal command that could be mapped to a key or be executed trough this function (there is actually a feautre request for that tough).

Slightly offtopic: I think it would be nice to remove static commands as a concept in the future. Mappable commands make sense but static commands are really just something that can be mapped to a key. With scheme that can be any function so what are now static commands could just be functions (that may or may not take arguments) that we export to scheme.

The scheme keymap code can then simply map arbitrary functions to keys (with some convenience macros). Our current static commands would just be functions (that is actually already implemented in the plugin system PR). Not really related but just a tought I had altough to keep the PostCommand hook useful we may want to retain the ability to give names to commands.